scripts/VisionSuite/houghCircles.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+"""
+houghCircles.py
+---------------
+
+Finds the soccerball using opencv's Hough circle transfrom
+
+Example code at for this library is at the bottom of the file and will execute
+if this file is run.
+
+"""
+import os, cv2
+import numpy as np
+from utils import HSVFilter
+
+def houghCircles(img,
+    visualize=False,
+    setThresholds=False,
+    bounds={"upperHSV":np.array([50,255,255]),"lowerHSV":np.array([45,180,30])}):
+
+    #Assume nothing is wrong
+    error = False 
+
+    """
+    # Create blurred img
+    #blur = cv2.medianBlur(img, 5)
+    hsv = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
+    blur = cv2.medianBlur(hsv, 9)
+
+
+    # Manually adjust thresholding
+    if setThresholds:
+        hf = HSVFilter(blur,bounds)
+        bounds = hf.createBounds()
+        print bounds
+
+    # Filter image
+    binary = cv2.inRange(blur, bounds["lowerHSV"],bounds["upperHSV"])
+    val = cv2.medianBlur(hsv[:,:,2], 5) 
+    grey = cv2.bitwise_and(val,val,mask = binary)  
+
+    """
+    
+
+    #***Apply Blur***
+    #average
+    #blur = cv2.blur(img, (9,9))
+    #gaussian
+    #blur = cv2.GaussianBlur(img,(9,9),0)
+    #median
+    blur = cv2.medianBlur(img, 9)
+    #bilaterial
+    #blur =cv2.bilateralFilter(img,7,11,11)
+
+    # Filter image
+    grey = cv2.cvtColor(blur, cv2.COLOR_BGR2GRAY) 
+
+    # Apply Hough circles transform
+    circles = cv2.HoughCircles(grey,cv2.cv.CV_HOUGH_GRADIENT,1,100,param1=50,param2=50,minRadius=20,maxRadius=250)
+    
+    if circles is not None:
+        #round values to ints
+        circles = np.uint16(np.around(circles[0]))
+        #TODO:pick correct one
+        x,y = circles[0][0],circles[0][1]
```

Comment:
I take it this always picks the first for now as its 'algorithm'. If that's correct, I'm actually surprised how well this works.